### ISSUE TITLE
Different F1 Score with Different batch_size

### ISSUE THREAD

**Issue by mossaab0** [Lightning-AI/torchmetrics#845]
## 🐛 Bug

When I change the batch size, the F1 score (`val_f1`) changes. I suspect that the F1 score reported is the average F1 score across all minibatches. The suspicion comes from the observation that TP, TN, FP and FN in the `batch_size=100` run are averaged over the 19 minibatches (compared with a single pass across the entire dataset with `batch_size=1900`). E.g., `val_tp = 12 = 228 / 19`.

### To Reproduce

```
class MyClassifier(pl.LightningModule):
    def __init__(self):
        super(MyClassifier, self).__init__()
        [...]
        self.accuracy = torchmetrics.Accuracy(multiclass=False)
        self.f1 = torchmetrics.F1(multiclass=False)
        self.cm = torchmetrics.ConfusionMatrix(2)

    def validation_step (self, batch, batch_idx=None, optimizer_idx = None):
        encoded_sentences, labels = batch
        logits = self._get_logits(encoded_queries)
        probs = self.sigmoid(logits)
        cm = self.cm(probs.detach(), labels)
        metrics = {f'{stage}_f1': self.f1(probs.detach(), labels),
                   f'{stage}_acc': self.accuracy(probs.detach(), labels),
                  f'{stage}_tn': cm[0][0],
                  f'{stage}_fp': cm[0][1],
                  f'{stage}_fn': cm[1][0],
                  f'{stage}_tp': cm[1][1]}
        self.log_dict(metrics, on_step=False, on_epoch=True, prog_bar=True, sync_dist=True, batch_size=len(labels))
        return metrics
```

![image](https://user-images.githubusercontent.com/2341730/154545757-741951a8-ca7d-4cbb-a9a1-3d814fbd2411.png)

### Expected behavior

F1 score should be the same regardless of the `batch_size`.

### Environment

- PyTorch Version (e.g., 1.0): 1.10.2
- OS (e.g., Linux): Linux
- How you installed PyTorch: pip
- Python version: 3.7.0
- CUDA/cuDNN version: 11.0
- GPU models and configuration: Tesla V100-SXM2

**Comment by SkafteNicki** [comment]
I assume this has to do with the argument `reduce_fx` in `self.log` by default being `mean`:
https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#log

Essentially, if you log the values returned by the metric objects instead of object itself, lightning will be the one doing the accumulation instead of torchmetrics, which may lead to the wrong result.
To counter this, you have two options:
* Either log the metric object itself, so instead of calling
`self.log("f1", self.f1(probs.detach(), labels)`
do
```
self.f1(probs.detach(), labels)
self.log("f1", f1)
```
* Alternatively, I recommend that you explicit logs the final computed value to make it very clear that torchmetrics should do the accumulation and not lightning. Something like this should work:
```python
class MyClassifier(pl.LightningModule):
    def __init__(self):
        super(MyClassifier, self).__init__()
        [...]
        self.accuracy = torchmetrics.Accuracy(multiclass=False)
        self.f1 = torchmetrics.F1(multiclass=False)
        self.cm = torchmetrics.ConfusionMatrix(2)

    def validation_step (self, batch, batch_idx=None, optimizer_idx = None):
        encoded_sentences, labels = batch
        logits = self._get_logits(encoded_queries)
        probs = self.sigmoid(logits)
        self.cm.update(probs.detach(), labels)
        self.f1.update(probs.detach(), labels)
        self.accuracy.update(probs.detach(), labels)

    def validation_epoch_end(self, outputs):
        acc = self.accuracy.compute()
        f1 = self.f2.compute()
        cm = self.cm.compute()
        metrics = {f'{stage}_f1': f1,
                   f'{stage}_acc': acc,
                  f'{stage}_tn': cm[0][0],
                  f'{stage}_fp': cm[0][1],
                  f'{stage}_fn': cm[1][0],
                  f'{stage}_tp': cm[1][1]}
        self.log_dict(metrics, prog_bar=True, sync_dist=True, batch_size=len(labels))
        return metrics
```

**Comment by SkafteNicki** [comment]
Also, you can find more info on this page:
https://torchmetrics.readthedocs.io/en/latest/pages/lightning.html
